examples/coexist_klog_v1/coexist_klog_v1.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"flag"
+
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+)
+
+type klogWriter struct{}
+
+func (kw klogWriter) Write(p []byte) (n int, err error) {
```

Comment:
this is clever, but I have a few questions:
* is it valid to assume every Write contains an entire message, where p[0] is the log header, or is Write ever called with partial lines or non-log-header lines?
* same question as https://github.com/kubernetes/klog/pull/142/files#r423834783 about depth of 6

Reply:
@liggitt yes. by the time it gets here it has the standard prefix for each line

yes the magic depth of 6 gets us the right file names. I don't see any other way to do this :(

Reply:
> is Write ever called with partial lines or non-log-header lines?

For example, logging a multi-line message, or a message longer than the default buffer size?

This turns the way klog/v2 makes `Write()` calls into an API... which is clever, but I could easily see causing problems in the future

Reply:
> the magic depth of 6 gets us the right file names. I don't see any other way to do this :(

do all calls to Write() occur at a constant distance from non-klog packages?

Reply:
yes they do

Reply:
Adding a test

Reply:
should we expose package constants (e.g. `OutputCallDepth`, `LoggerCallDepth`) that can be used for this? we could verify with tests they are correct, and if the klog internals change, we could adjust them as needed

Reply:
Ack will do that

Reply:
Done!